applications/High_availability_validator_setup.md

```diff
@@ -0,0 +1,182 @@
+
+### Milestone 2 - External service for permission granting
+
+Sendecond milestone introduces a microservice which will test dynamic switching of the permission granting, during the validators work. We will be able to test permission granting triggered whenever the block is produced, vote is made on the finalized block (GRANDPA) and message (`I'm online`) is sent for communicating liveness. Only the leader validator will be granted to run those actions. At worst the author may miss a slot.
+
+The service should contain only basic logic (e.g. return `true` for node that asked first and `false` for following ones).
+
+
+* **Estimated Duration:** 3 weeks
+* **FTE:** 1
+* **Costs:** 7,500 DAI
+
+| Number | Deliverable | Specification | 
+| ------------- | ------------- | ------------- |
+| **0a.** | License | Apache 2.0 / GPLv3 / MIT / Unlicense |
+| **0b.** | Documentation | We will provide inline documentation.|
+| **0c.** | Testing and Testing Guide | We will provide unit tests and the guidelines for running and testing it the scope. |
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 1. | Basic service | Create a microservice that accepts connections from the node. | 
+| 2. | Getting permission from microservice | Setup custom node repository and write PermissionResolver trait implementation for getting permission from microservice. |
+| 3. | Allow as optional | The choice of using an outside decision making agent for block submission should be configurable in the cli. |
+| 4. | Cleanup substrate code | Remove deprecated cli options.
+| 5. | Integration test | A dockerised setup that allows to run custom nodes network and microservice in order to show that created solution works. |
+
+### Milestone 3 - Raft based current leader selection
+
+Replace the dummy microservice as an infrastructure component with a TiKV cluster used for leader selection. 
+Each node should try to get authorship premission based on the KV state. Replace current microservice client with TiKV client and add corresponding logic.
+
+* **Estimated Duration:** 2 weeks
+* **FTE:** 1
+* **Costs:** 5,000 DAI 
+
+| Number | Deliverable | Specification | 
+| ------------- | ------------- | ------------- |
+| **0a.** | License | Apache 2.0 / GPLv3 / MIT / Unlicense |
+| **0b.** | Documentation | We will provide inline documentation.|
+| **0c.** | Testing and Testing Guide | We will provide unit tests and the guidelines for running and testing it the scope. |
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 1. | Run the necessary Raft services | Setup local dev infrastructure to run TiKV components in order to provide distributed KV store. |
+| 2. | Integrate a Raft client into the node | Replace previous logic with TiKV based one and modify configs to allow TiKV client to connect to Placement Drivers. | 
```

Comment:
```suggestion
| 2. | Integrate a Raft client into the node | Replace the previous logic with a TiKV based one and modify configs to allow the TiKV client to connect to Placement Drivers. | 
```

Reply:
> Placement Drivers

What's that?

Reply:
Placement Drivers (PD) is one of the components which stores metadata for the entire TiKV cluster. It is responsible for sending commands to the TiKV nodes. The minimum setup for PD is to manage three TiKV nodes.

Reply:
@mgralinski-bright could you incorporate this into the doc? other than that I'm willing to approve it as well.